src/core/handlers/RequestHandler.ts

```diff
@@ -200,7 +204,16 @@ export abstract class RequestHandler<
 
     // Clone the request instance before it's passed to the handler phases
     // and the response resolver so we can always read it for logging.
-    const mainRequestRef = args.request.clone()
+    if (!RequestHandler.#mainRequestRefCache.has(args.request)) {
+      RequestHandler.#mainRequestRefCache.set(
+        args.request,
+        args.request.clone(),
+      )
+    }
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const mainRequestRef = RequestHandler.#mainRequestRefCache.get(
+      args.request,
+    )!
```

Comment:
not the ideal way to do this, only showing how we can remove the max listener warnings by removing almost all cloning. this is also a big perf win